src/scenes/NftDetailPage/NftDetailPage.tsx

```diff
@@ -19,10 +19,13 @@ type Props = {
 };
 
 function NftDetailPage({ nftId }: Props) {
-  const { replace, back } = useRouter();
+  const { replace, back, query } = useRouter();
   const canGoBack = useCanGoBack();
 
   const username = window.location.pathname.split('/')[1];
+  const collectionId = query.collectionId;
+  const isFromCollectionPage =
+    globalThis?.sessionStorage?.getItem('prevPage') === `/${username}/${collectionId}`;
```

Comment:
same comment... we should brainstorm what the best way is to see where they've come from. we can ask terence next week.

Reply:
Sure! The current way of dealing kind of hacky things..

Reply:
Seems like we could refactor some of the navigation context stuff to store a stack of what routes we've been on so we could actually track this. without local storage.

Reply:
It's not blocking though, we can land this

Reply:
@tbezman I like this idea. If I'm not mistaken, currently we only store only number of stack in the context.